packages/runner/src/suite.ts

```diff
@@ -768,14 +770,15 @@ export function createTaskCollector(
     ) {
       const collector = getCurrentSuite()
       const scopedFixtures = collector.fixtures()
+      const context = { ...this }
       if (scopedFixtures) {
-        this.fixtures = mergeScopedFixtures(
-          this.fixtures || [],
+        context.fixtures = mergeScopedFixtures(
+          context.fixtures || [],
```

Comment:
`this` is shared between all `test` function as per `fn.apply(context, args)` in

https://github.com/vitest-dev/vitest/blob/dee7ce9928f065fd65b4e57bc8db3765a4679987/packages/runner/src/utils/chain.ts#L11-L18

This was causing `extendedTest` to track `this.fixtures` in a stateful way instead of suite level inheritance. Existing tests were probably relying on that behavior.

Now I removed this stateful mutation logic, so proper suite level fixture inheritance is required and that's done via `createSuiteCollector`'s `parentCollectorFixtures`.